columnar/src/column_index/mod.rs

```diff
@@ -159,6 +159,18 @@ impl ColumnIndex {
             }
         }
     }
+
+    pub fn get_docids_with_existing_value(
+        &self,
+        selected_docid_range: Range<u32>,
+        doc_ids: &mut Vec<u32>,
+    ) {
+        for doc_id in selected_docid_range {
```

Comment:
With the given implementation of get_docids_with_existing_value, I don't think the buffering is worth it.
@PSeitz may have a better insight on this.

I'd rather have a simple first implementation that does not fill a buffer with a dynamic horizon, but instead just seeks for the next doc that has a value in a dumb for loop.

The buffering could help if we have a more sophisticated implem of get_docs_with_existing_values.

Reply:
The buffering mainly reduced the overhead of single calls, e.g. dynamic dispatch, translating into value space and SIMD.

Only dynamic dispatch applies here, but since every doc is called separately, there's no gain here.

A push down of the check to the index implementation still may make sense, but we should back it with benchmarks and I would start with the simple version.